pkg/markdown/markdown.go

```diff
@@ -39,7 +39,20 @@ func WithTheme(theme string) glamour.TermRendererOption {
 		switch theme {
 		case "light", "dark":
 			if accessible {
-				return glamour.WithStyles(AccessibleStyleConfig(theme))
+				// Applying multiple glamour.TermRendererOption here requires a wrapper that applies each
+				// within glamour.NewTermRenderer() in Render() below.
+				stylesOption := glamour.WithStyles(AccessibleStyleConfig(theme))
+				chromaOption := glamour.WithChromaFormatter("terminal16")
+
+				return func(tr *glamour.TermRenderer) error {
+					if err := stylesOption(tr); err != nil {
+						return err
+					}
+					if err := chromaOption(tr); err != nil {
+						return err
+					}
+					return nil
+				}
```

Comment:
Clever, but I really don't like how this reads... I find it pretty confusing given this is returning a `glamour.TermRendererOption`. I guess it's a function of this call signature, but that requires implementation level knowledge of glamour to understand this code.

I think I might prefer to abstract this closure out to something separate, so we end up with something like

```suggestion
				return AccessibleStyles(theme)

// in the accessibility dir

func AccessibleStyles(theme string) glamour.TermRendererOption {
    // Applying multiple glamour.TermRendererOption here requires a wrapper that applies each
    // within glamour.NewTermRenderer() in Render() below.
    stylesOption := glamour.WithStyles(AccessibleStyleConfig(theme))
    chromaOption := glamour.WithChromaFormatter("terminal16")
    
    return func(tr *glamour.TermRenderer) error {
	if err := stylesOption(tr); err != nil {
		return err
	}
	if err := chromaOption(tr); err != nil {
		return err
	}
	return nil
    }
}
```

That way, we're separating the logic of "this is how you build the accessible theme using our implementation" from "select the right theme to use"

Reply:
> Clever, but I really don't like how this reads... I find it pretty confusing given this is returning a `glamour.TermRendererOption`. I guess it's a function of this call signature, but that requires implementation level knowledge of glamour to understand this code.
> 
> I think I might prefer to abstract this closure out to something separate, so we end up with something like
> 
> That way, we're separating the logic of "this is how you build the accessible theme using our implementation" from "select the right theme to use"

I was definitely conflicted in the implementation because of where / how we were applying the accessible theme in `WithTheme` versus in `Render` where this would allow us to have separate `TermRendererOption` that would be cleaner.

That said, I think at some level the reader digging into this has to understand `glamour` as long as its the dependency we use here.

Do you think further abstracting this could create a different type of confusion for reasoning about this behavior?